schema.go

```diff
@@ -366,6 +367,9 @@ func structNodeOf(t reflect.Type) *structNode {
 			fields[i].Tag.Get("parquet-key"),
 			fields[i].Tag.Get("parquet-value"),
 		})
+		if withId, ok := field.Node.(nodeID); ok {
+			field.id = withId.ID()
+		}
```

Comment:
This may not work if the node is wrapped multiple times; for example:

```go
node = parquet.Optional(parquet.FieldID(node, id))
```

did you consider adding the `ID` method directly on the `Node` interface?

Reply:
This part is guaranteed to work because  the node in question is coming from `makeNodeOf` which has this body

```go
...
	if optional {
		node = Optional(node)
	}
	if fieldID != nil {
		node = FieldID(node, *fieldID)
	}
	return node
```

`FieldID` wrap will always be after `Optional` in this case.

> did you consider adding the ID method directly on the Node interface?

Yes, initially but deemed it necessary because I was razor focused on just `SchemaOf` forgetting manual schema construction.

Seeing I confused you on the order of node wrapping, I think having `ID` on `Node` makes more sense now.

done a9c4cfd